tempesta_fw/classifier/frang.c

```diff
@@ -993,6 +1137,100 @@ frang_free_ct_vals(TfwCfgSpec *cs)
 }
 
 static int
+frang_register_fsm_resp(void)
+{
+	int r = tfw_gfsm_register_fsm(TFW_FSM_FRANG_RESP, frang_resp_handler);
+	if (r) {
+		TFW_ERR("\nfrang: can't register response fsm");
+		return r;
+	}
+
+	fsm_hook_resp_prio = tfw_gfsm_register_hook(TFW_FSM_HTTP,
+						    TFW_GFSM_HOOK_PRIORITY_ANY,
+						    TFW_HTTP_FSM_RESP_MSG_FWD,
+						    TFW_FSM_FRANG_RESP,
+						    TFW_FRANG_RESP_FSM_INIT);
+	if (fsm_hook_resp_prio < 0) {
+		TFW_ERR("\nfrang: can't register gfsm msg fwd hook");
+		tfw_gfsm_unregister_fsm(TFW_FSM_FRANG_RESP);
+		return fsm_hook_resp_prio;
+	}
+
+	return 0;
+}
+
+static int
+frang_parse_ushort(const char *s, unsigned short *out)
+{
+	int n;
+	if(tfw_cfg_parse_int(s, &n) || n < 1 || n > USHRT_MAX) {
+		TFW_ERR_NL("frang: http_resp_code_block: \"%s\" isn't a valid value\n", s);
```

Comment:
The coding style says: "However, never break user-visible strings such as printk messages, because that breaks the ability to grep for them". 
It will be fixed in the next commit though.

Reply:
That's true, so we try to keep error messages as small as possible and/or break lines on ungrepable points, `"%s"` in this case.